runtime/cudaq/cudaq.cpp

```diff
@@ -36,74 +36,71 @@ cudaq::MPIPlugin *getMpiPlugin(bool unsafe) {
-      g_plugin = std::make_unique<cudaq::MPIPlugin>(mpiLibPath);
+  if (g_plugin)
+    return g_plugin.get();
+
+  // Search priority:
+  //  (1) Environment variable take precedence (e.g., by running the
+  // activation script)
+  //  (2) Previously-activated custom plugin at its default location
+  //  (3) Built-in comm plugin (e.g., docker container or build from source
+  // with MPI)
+  const char *mpiLibPath = std::getenv("CUDAQ_MPI_COMM_LIB");
+  if (mpiLibPath) {
+    // The user has set the environment variable.
+    cudaq::info("Load MPI comm plugin from CUDAQ_MPI_COMM_LIB environment "
+                "variable at '{}'",
+                mpiLibPath);
+    g_plugin = std::make_unique<cudaq::MPIPlugin>(mpiLibPath);
+  } else {
+    // Try locate MPI plugins in the install directory
+    std::filesystem::path cudaqLibPath{cudaq::getCUDAQLibraryPath()};
+    // First, look for the previously-activated plugin in the
+    // `distributed_interfaces/` directory.
+    const auto distributedInterfacesDir =
+        cudaqLibPath.parent_path().parent_path() / "distributed_interfaces";
+    // Note: this file name must match the one defined in
+    // `activate_custom_mpi.sh`.
+    constexpr std::string_view activatedInterfaceLibFilename =
+        "libcudaq_distributed_interface_mpi.so";
+    const auto activatedInterfaceLibFile =
+        distributedInterfacesDir / activatedInterfaceLibFilename;
+    if (std::filesystem::exists(activatedInterfaceLibFile)) {
+      cudaq::info("Load MPI comm plugin from '{}'",
+                  activatedInterfaceLibFile.c_str());
+      g_plugin =
+          std::make_unique<cudaq::MPIPlugin>(activatedInterfaceLibFile.c_str());
     } else {
-      // Try locate MPI plugins in the install directory
-      std::filesystem::path cudaqLibPath{cudaq::getCUDAQLibraryPath()};
-      // First, look for the previously-activated plugin in the
-      // `distributed_interfaces/` directory.
-      const auto distributedInterfacesDir =
-          cudaqLibPath.parent_path().parent_path() / "distributed_interfaces";
-      // Note: this file name must match the one defined in
-      // `activate_custom_mpi.sh`.
-      constexpr std::string_view activatedInterfaceLibFilename =
-          "libcudaq_distributed_interface_mpi.so";
-      const auto activatedInterfaceLibFile =
-          distributedInterfacesDir / activatedInterfaceLibFilename;
-      if (std::filesystem::exists(activatedInterfaceLibFile)) {
-        cudaq::info("Load MPI comm plugin from '{}'",
-                    activatedInterfaceLibFile.c_str());
-        g_plugin = std::make_unique<cudaq::MPIPlugin>(
-            activatedInterfaceLibFile.c_str());
-      } else {
-        const auto pluginsPath = cudaqLibPath.parent_path() / "plugins";
+      const auto pluginsPath = cudaqLibPath.parent_path() / "plugins";
 #if defined(__APPLE__) && defined(__MACH__)
-        const std::string libSuffix = "dylib";
+      const std::string libSuffix = "dylib";
```

Comment:
Could possibly use `constexpr std::string_view` here to avoid constructing a string on the heap that is just data to be copied into another string object.